### PR TITLE
gcp-observability: remove unregistering stackdriverexporter

### PR DESCRIPTION
**Pull request by DNVindhya**
This PR removes unregistering `Stackdriver<Stats/Trace>Exporter` because `unregister` shutdowns the channel even before creating metric descriptors causing `RunTimeException`.

Added a null check for `sourceProjectId` for non-GCP instances.

CC @sanjaypujare

### CODE REVIEW

File: gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java [hunk 1]
```diff
@@ -148,6 +148,7 @@ static Map<String, String> getCustomTags(Map<String, String> customTags,
     ImmutableMap.Builder<String, String> tagsBuilder = ImmutableMap.builder();
     String sourceProjectId = locationTags.get("project_id");
     if (!Strings.isNullOrEmpty(destinationProjectId)
+        && !Strings.isNullOrEmpty(sourceProjectId)
```
Comment (sanjaypujare) [inline gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java:151]: This change is unrelated to the main intent of the PR, right?

Comment (DNVindhya) [reply gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java:151]: Yes. This is something I noticed while testing from my desktop.
Is it better to address this in a separate PR?

**Review by sanjaypujare — APPROVED** [review]
LGTM
